Std/Data/List/Lemmas.lean

```diff
@@ -2352,6 +2369,15 @@ theorem IsInfix.filter (p : α → Bool) ⦃l₁ l₂ : List α⦄ (h : l₁ <:+
   obtain ⟨xs, ys, rfl⟩ := h
   rw [filter_append, filter_append]; apply infix_append _
 
+theorem cons_prefix_iff : a :: l₁ <+: b :: l₂ ↔ a = b ∧ l₁ <+: l₂ := by
```

Comment:
Presumably this was already named like this in Mathlib, but I'm a little surprised this isn't `cons_prefix_cons`.

Reply:
Let's rename, adding a `@[deprecated] alias`.